pkg/apis/v1beta1/nodepool.go

```diff
@@ -124,7 +124,7 @@ type Budget struct {
 	// This is required if Schedule is set.
 	// This regex has an optional 0s at the end since the duration.String() always adds
 	// a 0s at the end.
-	// +kubebuilder:validation:Pattern=`^([0-9]+(m|h)+(0s)?)$`
+	// +kubebuilder:validation:Pattern=`^(([0-9]+(h|m)(0s)?)|([0-9]+h+[0-9]+m+(0s)?))$`
```

Comment:
I think there only needs to be one 0s at the end of the whole string, right before the $. It also seems like this means I could do 1h2h3m4m, which I don't think is what we want to do. 
```suggestion
	// +kubebuilder:validation:Pattern=`^(([0-9]+(h|m)(0s)?)|([0-9]+h+[0-9]+m+(0s)?))$`
```

Reply:
This should not allow for that kind of format. We are only allowing 1h/1m or 1h1m